src/extension/features/accounts/last-reconciled-date/index.js

```diff
@@ -0,0 +1,85 @@
+import { Feature } from 'toolkit/extension/features/feature';
+import { isCurrentRouteAccountsPage } from 'toolkit/extension/utils/ynab';
+import { controllerLookup } from 'toolkit/extension/utils/ember';
+import { getEntityManager } from 'toolkit/extension/utils/ynab';
+const YNAB_ACCOUNTS_HEADER_RIGHT = '.accounts-header-balances-right';
+const TK_LAST_RECONCILED_ID = 'tk-last-reconciled-date';
+
+export class LastReconciledDate extends Feature {
+  injectCSS() {
+    return require('./styles.css');
+  }
+
+  shouldInvoke() {
+    let { selectedAccountId } = controllerLookup('accounts');
+    return selectedAccountId && isCurrentRouteAccountsPage();
+  }
+
+  invoke() {
+    // Get the current account id and calculate the last reconciled date
+    let { selectedAccountId } = controllerLookup('accounts');
+    let latestDate = this._calculateLastReconciledDate(selectedAccountId);
+    let textToShow = 'NA';
+    if (latestDate) {
+      textToShow = ynab.YNABSharedLib.dateFormatter.formatDateExpanded(latestDate.utc());
+    }
+    // Retrieve or create the reconcile date container
+    let dateContainer = $(`#${TK_LAST_RECONCILED_ID}`);
+    if (!dateContainer || dateContainer.length === 0) {
+      $(YNAB_ACCOUNTS_HEADER_RIGHT).append(
+        `<div class="tk-accounts-header-last-reconciled">
+        <span id="${TK_LAST_RECONCILED_ID}">${textToShow}</span>
+        <div class="tk-accounts-header-last-reconciled-label">Last Reconciled Date</div>
+      </div>`
+      );
+    }
+
+    // Update the reconcile date in the element
+    dateContainer.text(textToShow);
+    this._setFeatureVisibility(true);
+  }
+
+  onRouteChanged() {
+    if (this.shouldInvoke()) {
+      this.invoke();
+    } else {
+      this._setFeatureVisibility(false);
+    }
+  }
+
+  observe(changedNodes) {
+    if (!this.shouldInvoke()) return;
+
+    // When the reconciled icon changes, reevaluate our date
+    if (changedNodes.has('is-reconciled-icon svg-icon lock')) {
+      this.invoke();
+    }
+  }
+
+  /**
+   * Calculate the last reconciled date
+   * @param {String} accountId The account id to get the reconciled date for
+   * @returns {Moment} the latest date, null otherwise
+   */
+  _calculateLastReconciledDate = accountId => {
```

Comment:
looks like we can replace all this logic with 

```javascript
controllerLookup('accounts').accountsService.selectedAccount.lastReconciledAt
```

Reply:
I tried this and for some reason, it was returning back null for some accounts when it was actually reconciled.
It was a hit or miss for me :/ 

I think it only accounts for the current month. I noticed if there was previously reconciled month and a new month began, the `lastReconciledAt` would be null, but if I reconciled again, it would populate. Suggestions?

Reply:
oh that's really lame :/ hmm i guess we'll just roll with this then!